include/nbl/builtin/hlsl/blit/blit.hlsl

```diff
@@ -0,0 +1,174 @@
+// Copyright (C) 2023 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_BLIT_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BLIT_INCLUDED_
+
+
+#include <nbl/builtin/hlsl/multi_dimensional_array_addressing/multi_dimensional_array_addressing.hlsl>
+#include <nbl/builtin/hlsl/blit/parameters.hlsl>
+#include <nbl/builtin/hlsl/blit/common.hlsl>
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+#include <nbl/builtin/hlsl/spirv_intrinsics/core.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace blit
+{
+
+// p in input space
+int32_t3 getMinKernelWindowCoord(NBL_CONST_REF_ARG(int32_t3) p, NBL_CONST_REF_ARG(float3) minSupport)
+{
+	return int32_t3(ceil(p - float3(0.5f, 0.5f, 0.5f) + minSupport));
+}
+
+template <
```

Comment:
1. its not nice to have an identifier that shadows a namespace name, so lets name this `compute_blit`

2. something to improve usability and style, change the `blit` from a function to a `struct` with a `template<> execute(...)` then you can spec certain template parameters and arguments on the `struct` and its `static compute_blit create(...)` methods while others on the `execute(...)` like the accessors used

Reply:
btw again its a bit weird that theres a `blit.hlsl` inside a `blit` folder in a `blit` namespace, so change the path to `nbl/builtin/hlsl/blit/compute_blit.hlsl` as well